net-mail/poppassd_ceti/poppassd_ceti-1.8.7.ebuild

```diff
@@ -0,0 +1,44 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit pam
+
+# Long ago it was just "poppassd", but upstream now seems to have
+# settled on "poppassd-ceti" (instead of "poppassd_ceti" or no suffix).
+MY_PN="poppassd-ceti"
+MY_P="${MY_PN}-${PV}"
+S=${WORKDIR}/${MY_P}
+
+DESCRIPTION="Password change daemon with PAM support"
+HOMEPAGE="https://github.com/kravietz/poppassd-ceti"
+SRC_URI="https://github.com/kravietz/${MY_PN}/releases/download/v${PV}/${MY_P}.tar.xz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+xinetd"
+
+DEPEND="sys-libs/pam"
+
+RDEPEND="${DEPEND}
+	xinetd? ( sys-apps/xinetd )
```

Comment:
Sorry to say, but if this doesn't have a function in this ebuild, it doesnt belong as a USE Flag. Use `optfeature` from eutils.eclass instead, or a-like. 
https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Policies#USE-Controlled_Optional_RDEPENDs
There's also insinto /etc/xinetd.d below, are you sure its optional?

Reply:
TLDR version: [x]inetd is essentially never the only way to run ANY package, even if for some packages it is "usually" the only way a package is used.  So it is hard to decide how it should be indicated in package dependencies, especially without some kind of global guidelines.

----
In a quick, informal survey / skim of various ebuilds that interact with inetd, I noticed there is a fair amount of inconsistency and strangeness with respect to how it is handled.  For example:

   * Background: Just by its very nature, generally speaking/in most cases, anything that is intended to be launched by one form of inetd could easily be launched by any other form.  In fact, inetd is usually not actually required at all, even if it is "typical".  It could be executed directly or via sudo or ssh instead, and just talk to it with its appropriate protocol via stdin/stdout.  There might be exceptions to this for various odd reasons, but I believe such exceptions are rare.
   * There is a virtual/inetd, but only a few packages depend on it.  More packages directly depend on sys-apps/xinetd instead, but in most cases the specificity is probably unnecessary.
   * Packages that depend on virtual/inetd generally still install /etc/xinetd.d/* config files, which would generally be ignored if the user elects to use some other inetd (or ssh or sudo or ...).
   * Clearly xinetd is the default (preferred) inetd (better configurability, ability to have a directory of independently-managed config files, etc), but it isn't the only option in the tree.
   * Most packages that use the xinetd USE flag do two things with it: add a conditional dependency (sometimes on sys-apps/xinetd, sometimes on virtual/inetd), and conditionally install the xinetd config file for the package.  (A few packages also tweak something else, but I didn't look carefully to see if any of them change compile-time options, vs only change default configuration files.  I suspect it is mostly default files...)
   * dev-vcs/cvs is an interesting case study.  It does not indicate a dependency on any inetd at all, but will install xinetd config files if cvs is compiled in a way that allows it to act as a server via the "server" USE flag.  (It looks like with the ebuild, cvs is always configured to support client mode, although I could have sworn that the lower level configure can actually setup a server-only cvs (but not as default), at least in the distant past.)  But even when server is enabled, it is (was? cvs is almost dead) more common to invoke cvs server mode via ssh, not via an inetd.
   * I believe there is a policy that systemd unit files are always installed, even if the systemd USE flag is off and systemd itself is masked.  Perhaps arguably xinetd config files are conceptually similar enough that maybe they should fall under the same policy?

I wonder if maybe some more guidelines or policies about inetd should be discussed and adopted across gentoo?  Some possibilities to consider include:

   * Only depend on virtual/inetd, unless there is some weird inherit dependency on xinetd specifically?
   * Install xinetd config files if they could work (see cvs), even if some other inetd is currently selected?
   * Avoid depending on any inetd at all unless that really is the only way the package is typically used (not ssh, sudo, or anything else)?
   * There is (or should be) a policy about this already, but if not, make sure that any xinetd config files are "disabled = yes" by default unless the user explicitly enables them.
   * Make an explicit exception to the "USE flag not used for optional dependencies" policy for the "xinetd" USE flag when a package is usually used with xinetd (easiest to configure), but might sometimes be used without it?

----
However, in the absence of any such policy adjustments, the three most likely ways to address your concerns with my poppassd_ceti ebuild might be either:

   * Eliminate any inetd dependency completely, but still install the xinetd config file.  Unconditionally elog a message about how poppassd has to run as root to work, and various ways to run it that way (including xinetd and sudo).
   * Or keep the USE-controlled dependency, but also make installation of the xinetd config file depend on the same flag.  This is the way the flag is used in most ebuilds that test the flag.  Perhaps depend on virtual/inetd instead of sys-apps/xinetd (not sure)?  Probably also elog a message in this case as well.
   * Or we could keep what poppassd_ceti's previous ebuild does, and unconditionally depend on xinetd.  But I've personally never run it that way - I have xinetd installed via dependencies like this, but don't actually start it or use it for anything.  My password change web page backend uses sudo instead.  It would be nice (but not critical) to be able to install poppassd_ceti without xinetd for use cases like mine.

Do you have a preferred strategy?

Reply:
In the absence of any guidelines or suggestions, this new PR version goes with my first strategy from near the bottom of the above long analysis.